pages/SingleArticlePage.jsx

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from "react";
+import { Link, useNavigate, useParams } from "react-router-dom";
+import { getArticleById } from "../api/api";
+import SingleArticleCard from "../components/SingleArticleCard";
+
+export default function SingleArticlePage() {
+  const { article_id } = useParams();
+  const [currArticle, setCurrArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    getArticleById(article_id).then((data) => {
+      setCurrArticle(data);
+      setIsLoading(false);
+    });
+  }, []);
+
+  if (isLoading) <h2>Loadin...</h2>;
```

Comment:
This loading pattern isn't rendering as you'd hoped. On a slow connection I get no information